yt_dlp/extractor/microsoftembed.py

```diff
@@ -45,21 +45,17 @@ def _real_extract(self, url):
         timestamp = traverse_obj(
             metadata, ('snippet', 'activeStartDate'))
 
-        subtitles = {}
-
-        for lang, data in (traverse_obj(metadata, 'captions') or {}).items():
-            subtitles[lang] = [{
+        subtitles = {
+            lang: [{
                 'url': data.get('url'),
                 'ext': 'vtt'
-            }]
-
-        thumbnails = []
+            }] for lang, data in traverse_obj(metadata, 'captions', default={}).items()
+        }
 
-        for thumbnail_size, data in (traverse_obj(metadata, ('snippet', 'thumbnails')) or {}).items():
-            thumbnails.append({
-                'url': data.get('url'),
-                'http_headers': data.get('link')
-            })
+        thumbnails = [{
+            'url': data.get('url'),
+            'http_headers': data.get('link')
+        } for thumbnail_size, data in traverse_obj(metadata, ('snippet', 'thumbnails'), default={}).items()]
```

Comment:
ah, right. I didnt notice thumbnail_size is unused. Even if we don't want to use dict in traverseobj, this can atleast be simplified to:
```suggestion
        thumbnails = [{
            'url': data.get('url'),
            'http_headers': data.get('link')
        } for data in traverse_obj(metadata, ('snippet', 'thumbnails', ...))]
```

Reply:
Or, if you can get `height` from `thumbnail_size`, then do that instead and ignore this

Reply:
The thumbnails in the metadata look like

```
"thumbnails": {
      "extrasmall": {
        "height": 0,
        "width": 0,
        "assetId": "RWL7Ju",
        "url": "http://img-prod-cms-rt-microsoft-com.akamaized.net/cms/api/am/imageFileData/RWL7Ju?ver=cae5",
        "link": {
          "href": "/vhs/api/videos//thumbnails/extrasmall",
          "method": "GET",
          "rel": "self"
        }
      },
      "small": {
[...]
      ```
      
      in this case the height/width are always 0... but there is the keys to sort by in some way.

Reply:
simplified in commit 4fd9daa4d97f2555554a7c7694c86371eca40d98

Reply:
`height` and `width` should still be supplied, they contain values for example in `RWejb0`.